apis/apps/v1alpha1/cluster_types.go

```diff
@@ -913,19 +912,76 @@ type ServiceRef struct {
 	// +optional
 	Cluster string `json:"cluster,omitempty"`
 
+	// Specifies the cluster to reference.
+	//
+	// +optional
+	ClusterRef *ServiceRefClusterSelector `json:"clusterRef,omitempty"`
+
 	// The service descriptor of the service provided by external sources.
 	//
-	// When referencing a service provided by external sources, the ServiceDescriptor object name is required to
-	// establish the service binding.
+	// When referencing a service provided by external sources, a ServiceDescriptor object is required to establish
+	// the service binding.
 	// The `serviceDescriptor.spec.serviceKind` and `serviceDescriptor.spec.serviceVersion` should match the serviceKind
-	// and serviceVersion defined in the service reference declaration in the ClusterDefinition.
+	// and serviceVersion declared in the definition.
 	//
 	// If both Cluster and ServiceDescriptor are specified, the Cluster takes precedence.
 	//
 	// +optional
 	ServiceDescriptor string `json:"serviceDescriptor,omitempty"`
 }
 
+type ServiceRefClusterSelector struct {
+	// The name of the cluster to reference.
+	//
+	// +kubebuilder:validation:Required
+	Cluster string `json:"cluster"`
+
+	// The service to reference from the cluster.
+	//
+	// +optional
+	Service *ServiceRefServiceSelector `json:"service,omitempty"`
+
+	// The credential (SystemAccount) to reference from the cluster.
+	//
+	// +optional
+	Credential *ServiceRefCredentialSelector `json:"credential,omitempty"`
+}
+
+type ServiceRefServiceSelector struct {
+	// The name of the component where the service resides in.
+	//
+	// It is required when referencing a component service.
+	//
+	// +optional
+	Component string `json:"component,omitempty"`
+
+	// The name of the service to reference.
+	//
+	// Leave it empty to reference the default service. Set it to "headless" to reference the default headless service.
+	//
+	// +kubebuilder:validation:Required
+	Service string `json:"service"`
+
+	// The port name of the service to reference.
+	//
+	// If not specified, the first port will be used by default.
+	//
+	// +optional
+	Port string `json:"port,omitempty"`
```

Comment:
Defining Port as type IntOrString would be better?

Reply:
It's the name of the port in services, not a port value which can be in type of IntOrString.